### DISCUSSION TITLE
Volumes dashboard shows duplicate entries when using predict_linear function

### DISCUSSION THREAD

**Discussion by s-kuchi** [NetApp/harvest#2518] [Q&A]
When I add a query (shown below), I see multiple entries as shown in the screenshot. Any idea how to fix that?

<img width="1334" alt="Screenshot 2023-11-29 at 1 32 46 PM" src="https://github.com/NetApp/harvest/assets/133256656/0c18581b-04e3-4530-a27d-34156f0775d1">

```
predict_linear(volume_size_used_percent{datacenter=~"$Datacenter",cluster=~"$Cluster",svm=~"$SVM",volume=~"$Volume"}[60d],7776000)
```

**Comment by cgrinds** [comment]
hi @s-kuchi did you see multiple queries before adding a query? Are you modifying the existing panel or creating a new one? 

The two rows you are pointing to in the original image, look different but hard to tell since the column names are truncated. What if you run the same query in Prometheus and skip Grafana, do you see duplicate entries?

FWIW I added the same query to the `Volume Dashboard > Volume Table` and the query works without duplication. 

<img width="868" alt="image" src="https://github.com/NetApp/harvest/assets/242252/cef2d616-6fce-40bb-84be-c7b19c3ba092">

To make this work you need to select the value (Value #K for me) in the `Filter by name` section in the `Transform` tab like so

<img width="714" alt="image" src="https://github.com/NetApp/harvest/assets/242252/bcce95b0-8d2d-4c05-afa5-c53009dc67d2">

**Comment by s-kuchi** [comment]
Hi @cgrinds . If I "hide" the query, I do not see any duplication. I ran the same query in Prometheus and I see 2 values and the only difference between them is the node information.

![Screenshot 2023-11-29 at 2 12 18 PM](https://github.com/NetApp/harvest/assets/133256656/cd7e4324-1dec-47fe-8291-31ac90608ef1)
![Screenshot 2023-11-29 at 2 20 40 PM](https://github.com/NetApp/harvest/assets/133256656/f6ded468-6037-4759-a909-ba3c156e2546)
![Screenshot 2023-11-29 at 2 20 03 PM](https://github.com/NetApp/harvest/assets/133256656/2b7c65fa-b8e1-4385-aa7d-92fab4260940)
![Screenshot 2023-11-29 at 2 18 40 PM](https://github.com/NetApp/harvest/assets/133256656/ca4e1df1-33d6-4fcb-9d7d-ee85a1a5c3ab)

**Comment by cgrinds** [comment]
Thanks for the screenshots. What version of Harvest are you using and which collector, Zapi or Rest? I believe you are hitting issue #1360 which was fixed in March and was released in 23.05.

Can you try upgrading to 23.11 and see if you see the duplicates?

Also do you have much experience with `predict_linear`? If a simple linear regression works well for time to full, we should add that to the out-of-the-box dashboards. Thanks for bringing that to our attention. Looks like you picked 90 days in the future - any experiments with other values or the first one you started with?

**Reply by s-kuchi** [reply]
I am running 
harvest version 23.08.0-1 (commit 874a1f87) (build date 2023-08-21T08:36:53-0400) linux/amd64
checking GitHub for latest... newer version v23.11.0 available at https://github.com/NetApp/harvest/releases/latest

and ZAPI

{"level":"info","Poller":"wdc_netapp_a_fas","collector":"Zapi:Volume","instances":170,"apiMs":283,"parseMs":49,"calcMs":0,"pluginMs":94,"metrics":8439,"caller":"collector/collector.go:483","t
ime":"2023-11-29T03:49:49-05:00","message":"Collected"}

I can try upgrading it to 23.11

I have just started using predict_linear, so i can't speak to its accuracy, I have 2 queries, one for 90 days and one for 180 days. I am curious to see their accuracy since in my case a 60 day pattern can be used (in theory) to predict since I saw a repeated usage pattern.  IF you can use the same query ActiveIQ's trend graph (below) uses (not sure what query can reproduce that ), and use the same for the out-of-box dashboard, that will be helpful. Just a thought. 

![image](https://github.com/NetApp/harvest/assets/133256656/445698b4-afcc-4dea-97c9-5cd3c115ccea)

**Reply by cgrinds** [reply]
@rahulguptajss had another idea - is it possible that a node went down within the window you are querying (90 days), and the volume moved to another node?

**Reply by s-kuchi** [reply]
@cgrinds : I think what @rahulguptajss said makes sense. Any ideas how to write a query to correct that?
